app/models/race.rb

```diff
@@ -91,6 +93,20 @@ def maybe_start!
     return if started? || entries.where(readied_at: nil).any? || entries.count < 2
 
     update(started_at: Time.now.utc + 20.seconds)
+
+    # Schedule ghost splits and finishes
+    entries.ghosts.find_each do |entry|
+      entry.update(finished_at: started_at + (entry.run.duration(Run::REAL).to_sec))
+      Api::V4::RaceBroadcastJob.set(wait_until: started_at + entry.run.duration(Run::REAL).to_sec).perform_later(
+        self, 'race_entries_updated', 'A ghost has finished'
```

Comment:
I feel like ghosts should have their own status, and not just be entries updating since it isn't a true entrant.  That way 3rd parties can ignore the ghosts if they choose too.  What do you think?

Reply:
I actually like the idea of ghosts not needing any special code client-side, except maybe a styling change if `entry.ghost == true`. But I'm open to hear any practical advantages of separating them!

For disadvantages: if ghosts had a separate status, that's another event clients have to manually listen for (and might opt not to) and make display changes based on. But ultimately ghosts should be mixed into the list of real runners, so then you basically end up listening for 2 events that have the same effect (re-render the list of runners).

However I do think the extreme version of this is worth considering, where ghosts don't send any events at all, and are just static fields that point to the run. The clients would look at the times in the run and time their display changes correctly so that the splits and finish times happen when they are supposed to.

I think that extreme is a cleaner concept and has a nice offline-friendly mode, but still requires a lot more work from clients to ultimately display something that could have just behaved as a runner the whole time, so not worth it overall.

Reply:
I feel like going the extreme route where they just get the run is not very "integrated".  As a 3rd party consumer I would feel like it's something that is just tacked on so I don't think that is a great way to go personally, but I'm sure someone else may disagree.

I don't really have any clear cut advantages to having them be separate, it's just a sort of gut feeling I have.